packages/app-mobile/components/NotesBar.tsx

```diff
@@ -0,0 +1,282 @@
+			<View style={[styles.topContainer, styles.horizontalFlex]}>
+				{titleComp}
+				{closeButtonComp}
+			</View>
+			{dividerComp}
+		</View>
+	);
+
+	const handleQuerySubmit = async () => {
+		if (!query) {
+			setNotes(props.items);
+			return;
+		}
+
+		let searchQuery = query.trim();
+
+		if (searchQuery === '') {
+			setNotes(props.items);
+			return;
+		}
+
+		searchQuery = searchQuery.toLowerCase();
+
+		// Find notes in folder using only note title
+		const result = props.items.filter(item => {
+			let noteTitle = Note.displayTitle(item);
+			noteTitle = noteTitle.toLowerCase();
+
+			return noteTitle.includes(searchQuery);
+		});
+
+		setNotes(result);
+	};
+
+	const theme = themeStyle(themeId);
+
+	const searchInputComp = (
+		<View style={[styles.horizontalFlex, styles.searchInput]}>
+			<Icon name='search' style={[styles.top, styles.searchIcon]}/>
+			<TextInput style={[styles.top, styles.nativeInput]} placeholder='Search' onChangeText={setQuery} value={query} onSubmitEditing={handleQuerySubmit} placeholderTextColor={theme.dividerColor} />
+		</View>
+	);
+
+	const inputGroupComp = (
+		<View style={{ width: '100%' }}>
+			<View style={[styles.padding, styles.horizontalFlex, styles.inputGroup]}>
+				{searchInputComp}
+				{addNoteButtonComp}
+				{addTodoButtonComp}
+			</View>
+			{dividerComp}
+		</View>
+	);
+
+	let flatListRef: any = React.useRef(null);
+
+	const memoizedRenderItem = React.useCallback(({ item }: { item: any }) => {
+		if (item.is_todo) {
+			return <NotesBarListItem note={item} todoCheckbox_change={props.todoCheckbox_change} />;
+		} else {
+			return <NotesBarListItem note={item} />;
+		}
+	}, [props.todoCheckbox_change]);
+
+	const NotesBarListComp = (
+		<FlatList
+			data={notes}
+			renderItem={memoizedRenderItem}
+			keyExtractor={(item: any) => item.id}
+			getItemLayout={(data, index) => (
+				{
+					length: data.length,
+					offset: (theme.fontSize + styles.padding.paddingTop + styles.padding.paddingBottom) * index,
+					viewOffset: (theme.fontSize + styles.padding.paddingTop + styles.padding.paddingBottom),
+					index,
+				}
+			)}
+			ref={(ref: any) => { flatListRef = ref; }}
+		/>
+	);
+
+	// Scroll the notesbar to selected note item after rendering
+	React.useEffect(() => {
+		const selectedItemIndex = notes.findIndex(item => item.id === props.selectedNoteId);
+		if (selectedItemIndex >= 0) {
+			flatListRef.scrollToIndex({ index: selectedItemIndex });
+		}
+	});
+
+	// Update the notesbar when a note item changes
+	React.useEffect(() => {
+		setNotes(props.items);
+	}, [props.items]);
+
+	return (
+		<View style={styles.container}>
+			{topComp}
+			{inputGroupComp}
+			{ NotesBarListComp }
+		</View>
+	);
+}
+
+const NotesBar = connect((state: State) => {
+	return {
+		themeId: state.settings.theme,
+		items: state.notes,
+		activeFolderId: state.settings.activeFolderId,
+		selectedFolderId: state.selectedFolderId,
+		selectedNoteId: state.selectedNoteIds[0],
+		settings: state.settings,
```

Comment:
Please only pass the settings that you need, otherwise it can trigger many unnecessary re-renders. And in fact do you even need this props? It doesn't look like you're using it.

Reply:
Alright. No, I'm not using the settings props, I'll remove it.